CIP-0020/README.md

```diff
@@ -76,35 +86,185 @@ Example for multiple messages/comments/memos:
          }
 }
 ```
-Example above in **Daedalus** currently (could be improved if CIP is implemented):
-![image](https://user-images.githubusercontent.com/47434720/121822100-85b38a80-cc9d-11eb-9d13-1869746a69b2.png)
 
-## Some Integration examples
+&nbsp;<p>
+
+# Specification - Encrypted message
+
+This is an addition to the original CIP-0020, which is active since mid 2021 and widely used across many entities. It is fully backwards compatible and requires no changes in existing tools, explorers, wallets. 
+
+Why do we need this? Metadata is sent unencrypted and in plaintext over the networks, a 3rd-party or man-in-the-middle can easily collect data such as bank-account-numbers, email-addresses, etc. out of such messages. With even a simple encryption of such a message - and publicly known passphrase - it is much more complicated for the man-in-the-middle listener to collect data on the fly.
+
+The specification update for encrypted messages takes advantage of the simple original design, which is leaving room for additional json-keys not affecting the parsing of the content at all. The only outcome if a receiver does not process the encrypted content is, that the encrypted message is shown instead of an maybe autodecrypted one. But even the encrypted base64 strings fit into the max. 64char long string restriction. So it does not break any tools. More on the autodecryption later. 
+
+### Format:
+``` 
+{ 
+  "674":
+         { 
+           "enc": "<encryptionmode>",
+           "msg": 
+                  [ 
+                    "base64-string 1", "base64-string 2", "base64-string 3" ...
+                  ]
+         }
+}
+```
+The format is identical to the normal one, with a simple addition of the `enc` (encryptionmode) entry.
+
+The value given in the `enc` field references an entry in the [Encryption Mode Json](cip0020-encryption-modes.json) file, which collects different encryption methods and there parameters. Starting with a simple implementation named `basic`. This reference file is just a collection and should be seen as a look-up-file about the methods. This file can also be updated and extended with new encryption methods - like with public/private key encryption - easily in future updates.
+
+  &nbsp;<p>
+  
+### Encryption modes:
+
+* **plain** - no encryption at all
+``` json
+   "plain": {
+      "desc": "plaintext, no encryption" }
+```
+This is not really an encryption mode, but included as a backwards compatible entry to signal this message as an unencrypted one. The entry is not needed and fully optional for unencrypted messages.
+
+* **basic** - aes-256-cbc salted symmetric encryption via passpharse (+default passphrase)
+
+Lets list the entry from the [Encryption Mode Json](cip0020-encryption-modes.json) file first for the `basic` method:
+``` json
+   "basic": {
+      "desc": "symmetrical encryption via openssl and a passphrase (default=cardano)",
+      "type": "openssl",
```

Comment:
Hey @SebastienGllmt thx for the feedback!

The current format of this "lookup" file was choosen as a simple first start. It was not intended to be compatible for all possible future updates with all kinds of different encryption types. It could also be a simple text description file. It is ment currently as a place to collect the key parameters. 

I agree that CIP8 has some elements in it, but is to complicated to act as an easy upgrade to the existing transaction message/memo/comment format imo. When we started with CIP-0020 the intention was to keep it simple so many can implement it, but also leave it open enough to upgrade it. For the CIP8 style encryption, its ChachaPoly instead of AES256 for the method via passphrase. But CIP8 is oriented in signing in first place. Going with this format would make it totally incompatible with the current CIP-0020 format.

Let me think about it...

Reply:
What I like about this update is the simplicity of implementing openssl encryption for wallets. This isn't designed to be the end-all-be-all security solution for privacy on the blockchain. There's better options and upcoming Midnight for that. This is just so your receipt for a purchase at seedywebsite.com isn't in cleartext for everyone to see.